libfwupdplugin/fu-udev-device.c

```diff
@@ -1108,6 +1108,93 @@ fu_udev_device_set_physical_id (FuUdevDevice *self, const gchar *subsystems, GEr
 #endif
 }
 
+/**
+ * fu_udev_device_set_logical_id:
+ * @self: A #FuUdevDevice
+ * @subsystems: A subsystem string, e.g. `pci,usb`
+ * @error: A #GError, or %NULL
+ *
+ * Sets the logical ID from the device subsystem. Plugins should choose the
+ * subsystem that is "deepest" in the udev tree, for instance choosing 'usb'
+ * over 'pci' for a mouse device.
```

Comment:
your examples here aren't used, maybe make it choosing `hid` over `usb`

Reply:
Fixed. I've also done `s\subsystems\subsystem` as I don't think the fallback makes sense for the logical ID. You know what you're looking for, unlike the physical ID.